packages/grid/backend/backend.dockerfile.dockerignore

```diff
@@ -0,0 +1,67 @@
+# Syft
+tests/
+*.md
```

Comment:
I think this would be omitted as docker always looks for the .dockerignore in the build context directory

In the case of backend container, the build context is: `packages`
we could place it under packages folder

Reply:
See https://github.com/OpenMined/PySyft/pull/8605#discussion_r1597470335

Reply:
Correct, referring to the same document, the .dockerignore file is looked for in the root directory of context, in the case of backend build, since the context in packages directory, I think it should be present inside packages directory instead of packges/grid/backend

Reply:
![Uploading Screenshot 2024-05-14 at 12.39.08 PM.png…]()

Reply:
![Screenshot 2024-05-14 at 12 39 35 PM](https://github.com/OpenMined/PySyft/assets/43314053/fddd8b92-a14e-4ea6-b208-eb9aa3876574)

Reply:
Because the context keeps on changing for different build processes. In some it's packages/ in some it's application-specific like frontend/ & seaweedfs/. So this makes stuff way more clear no? So better to keep it next to the dockerfile so context can be anywhere as long as the paths are correct.

Reply:
Its only in `/package/` for the `backend` because it copies syft directly into the container. We could "build" a syft wheel first and then copy that in and install it instead and then the context path could be `backend`.

Reply:
Correct, since for backend the context is in `packages` but we have the .dockeringore inside `packagres/grid/backend` as docker looks for .dockerignore in the root directory of the context, so It would not take into account the `backend.dockerfile.dockerignore`  , so we could place this under `packages`

Reply:
> We could "build" a syft wheel first and then copy that in and install it instead and then the context path could be backend.

that would increase the build time so all gains would be lost :( And then we have the issue with torch where the cuda libs might be bundled with the wheel.

> ... so It would not take into account the backend.dockerfile.dockerignore , so we could place this under packages 

_"A Dockerfile-specific ignore-file takes precedence over the .dockerignore file at the root of the build context if both exist."_ https://docs.docker.com/build/building/context/#filename-and-location

The goal here is to keep things predictable and follow a pattern and have ignores next to their dockerfiles than have them spread over at different places based on their context.

Reply:
That statement takes only about precedence over multiple dockerignore files
But the dockerignore itself would not be considered , if it is not in the root directory of the context.

Reply:
Tested this on chat and ignore is working as expected - marking this as resolved